apps/dotcms-ui/src/app/view/components/dot-theme-selector-dropdown/dot-theme-selector-dropdown.component.ts

```diff
@@ -48,15 +48,17 @@ export class DotThemeSelectorDropdownComponent
     ) {}
 
     ngOnInit(): void {
-        this.currentSiteIdentifier = this.siteService.currentSite.identifier;
+        setTimeout(() => {
+            this.currentSiteIdentifier = this.siteService.currentSite.identifier;
+        }, 100);
     }
 
     ngAfterViewInit(): void {
         if (this.searchInput) {
             fromEvent(this.searchInput.nativeElement, 'keyup')
                 .pipe(debounceTime(500))
-                .subscribe((keyboardEvent: KeyboardEvent) => {
-                    this.getFilteredThemes(keyboardEvent.target['value']);
```

Comment:
not need to send the value entered on every debounced input, once that the search is triggered, the value from the input will be collected

this is part of performance improvement to reduce calls to the api